firmware/library/L2_Utilities/utility_test.cpp

```diff
@@ -0,0 +1,104 @@
+#include "L2_Utilities/utility.hpp"
+#include "L5_Testing/testing_frameworks.hpp"
+
+TEST_CASE("Testing Random Functions", "[random]")
+{
+    SECTION("SetRandomSeed Consistency")
+    {
+        utility::SetRandomSeed();
+        uint32_t randomtester1 = std::rand() % 100;
+        utility::SetRandomSeed();
+        uint32_t randomtester2 = std::rand() % 100;
+        CHECK(randomtester1 == randomtester2);
+    }
+
+    SECTION("Random")
+    {
+        utility::SetRandomSeed();
+        uint32_t randomtester1 = std::rand() % 100;
+        utility::SetRandomSeed();
+        uint32_t randomtester2 = utility::Random(0, 99);
+        CHECK(randomtester1 == randomtester2);
+    }
+}
+
+TEST_CASE("Testing Bit Manipulations", "[Bit Manipulation methods]")
```

Comment:
keep the [] test names short. For example this should just be `[bit manipulation]`